src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -0,0 +1,273 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { RemoteCheckoutActionCreator } from '../../../remote-checkout';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { AffirmAddress, AffirmDiscount, AffirmItem, AffirmRequestData, SCRIPTS_DEFAULT } from './affirm';
+import affirmJs from './affirmJs';
+declare let affirm: any;
+
+export default class AffirmPaymentStrategy implements PaymentStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator
+    ) { }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(options.methodId, options.gatewayId);
+
+        if (!paymentMethod ) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { testMode } = paymentMethod.config;
+        const { publicKey } = paymentMethod.initializationData;
+
+        affirmJs(publicKey, this._getScriptURI(testMode));
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const paymentId = payload.payment && payload.payment.methodId;
+        const { useStoreCredit } = payload;
+        const state = this._store.getState();
+        const config = state.config.getStoreConfig();
+
+        if (!paymentId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        if (!config) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        return this._store.dispatch(this._remoteCheckoutActionCreator.initializePayment(paymentId, { useStoreCredit }))
```

Comment:
I guess the reason why you have to post `useStoreCredit` via the remote endpoint is so that you can persist that setting the session and retrieve it back when you're redirected back. As I said, we need to reconsider this approach. It's merely a temporary workaround for AfterPay and we should aim to not replicate it for Affirm. Plus there are other reasons.